src/applications/vaos/appointment-list/redux/actions.js

```diff
@@ -389,6 +400,33 @@ export function fetchPastAppointments(startDate, endDate, selectedIndex) {
   };
 }
 
+export function fetchRequestDetails(id) {
+  return async (dispatch, getState) => {
+    const state = getState();
+    const { appointmentDetails, requestMessages } = state.appointments;
+    const pendingAppointments = selectPendingAppointments(state);
+    const request =
+      appointmentDetails[id] || pendingAppointments?.find(p => p.id === id);
+
+    dispatch({
+      type: FETCH_REQUEST_DETAILS,
+    });
+
+    if (request) {
+      dispatch({ type: FETCH_REQUEST_DETAILS_SUCCEEDED, request, id });
+    } else {
+      // TODO: fetch single appointment
+    }
+
+    const parsedId = parseFakeFHIRId(id);
+    const messages = requestMessages?.[parsedId];
+
+    if (!messages) {
+      dispatch(fetchRequestMessages(parsedId));
+    }
+  };
+}
```

Comment:
Set up this action in preparation for when we eventually do fetch appointment single details.  Fetched details will be stored in as hash, `appointments.appointmentDetails`.  For now, this action will check the hash to see if we've already fetched this appointment, or check `pendingAppointments`.